src/main/java/com/lothrazar/cyclicmagic/util/UtilPlaceBlocks.java

```diff
@@ -50,54 +50,50 @@ public static boolean placeStateSafe(World world, @Nullable EntityPlayer player,
       return false;
     }
     IBlockState stateHere = null;
-    if (player != null && PermissionRegistry.hasPermissionHere(player, placePos) == false) {
+    if (player != null && !PermissionRegistry.hasPermissionHere(player, placePos)) {
       return false;
     }
-    if (world.isAirBlock(placePos) == false) {
+    if (!world.isAirBlock(placePos)) {
       // if there is a block here, we might have to stop
       stateHere = world.getBlockState(placePos);
       if (stateHere != null) {
         Block blockHere = stateHere.getBlock();
-        if (blockHere.isReplaceable(world, placePos) == false) {
+        if (!blockHere.isReplaceable(world, placePos)) {
           // for example, torches, and the top half of a slab if you click
           // in the empty space
           return false;
         }
         // ok its a soft block so try to break it first try to destroy it
         // unless it is liquid, don't try to destroy liquid
-        //blockHere.getMaterial(stateHere)
+        // blockHere.getMaterial(stateHere)
         if (stateHere.getMaterial() != Material.WATER && stateHere.getMaterial() != Material.LAVA) {
-          boolean dropBlock = true;
           if (!world.isRemote) {
-            world.destroyBlock(placePos, dropBlock);
+            world.destroyBlock(placePos, true);
           }
         }
       }
     }
+    if (placeState.getBlock() instanceof BlockLeaves) { //dont let them decay
+      placeState = placeState.withProperty(BlockLeaves.DECAYABLE, false);
+    }
     boolean success = false;
     try {
       // as soon as i added the try catch, it started never (rarely) happening
-      // we used to pass a flag as third argument, such as '2'
-      // default is '3'
-      // if (!world.isRemote) {
-      if (placeState.getBlock() instanceof BlockLeaves) {//dont let them decay
-        placeState = placeState.withProperty(BlockLeaves.DECAYABLE, false);
+      // flags specifies what to update, '3' means notify client & neighbors
+      // isRemote to make sure we are in a server thread
+      if (!world.isRemote) {
+        success = world.setBlockState(placePos, placeState, 3); // returns false when placement failed
```

Comment:
so placement only happens server side now.  I think this change would mean that sound effects of block placements will not be played. I can see it commented out on the left "  // if (!world.isRemote) {" so i must have removed it for a reason.  so id leave this check out unless you have some reason i havent thought of

Reply:
you can't do world changes on the client.. probably the reason this whole thing is in a try in the first place, because this would crash when you run setBlockState on the client.. if success you are already playing the placement sound to all players in the area via your `playSoundPlaceBlock` helper class